rpcs3/Emu/System.cpp

```diff
@@ -2893,8 +2893,8 @@ void Emulator::Kill(bool allow_autoexit, bool savestate)
 
 			if (after_kill_callback)
 			{
-				after_kill_callback();
-				after_kill_callback = nullptr;
+				const auto callback = std::move(after_kill_callback);
+				callback();
```

Comment:
I don't see the purpose here.
This looks like a hazard. I don't think the moved object is guaranteed to be reset to default.

Reply:
It's also much easier to understand the original code

Reply:
I just checked and it is guaranteed to be empty afterwards.
after_kill_callback may contain a call to BootGame which may call Emu.Stop internally on error.

Reply:
iirc the only object that has no such guarentee in standard library is std::basic_string.

Reply:
I think it should at least have a comment and maybe the nullptr assignment anyway just to make the intentions clear.

Reply:
I added comment yes.